### PR TITLE
Add a new metric PENDING_TOPIC_LATENCY and record PRODUCE_ENCODE accurately

### LINKED ISSUE #1418: [BUG] Encode metric is not exactly the encode time
**Describe the bug**
Currently, the `kop_server_PRODUCE_ENCODE` metric is not exactly the encode time, we can see it's updated here:

https://github.com/streamnative/kop/blob/cc4ba412c5e776242de3d7b4e0ff732e9bbf299a/kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/storage/PartitionLog.java#L206-L207

There are two encoding related operations that need monitoring, see `[1]` and `[2]` in the following code.

```java
        final long beforeRecordsProcess = time.nanoseconds();
        try {
            final LogAppendInfo appendInfo = analyzeAndValidateRecords(records); // [1]
            /* ... */
            final Consumer<Optional<PersistentTopic>> persistentTopicConsumer = persistentTopicOpt -> {
                /* ... */
                final EncodeResult encodeResult = entryFormatter.encode(encodeRequest); // [2]
                encodeRequest.recycle();
                requestStats.getProduceEncodeStats().registerSuccessfulEvent(
                        time.nanoseconds() - beforeRecordsProcess, TimeUnit.NANOSECONDS);
```

However, the `persistentTopicConsumer` won't be triggered until the associated pending future (see the following code) of  is completed in `PendingTopicFutures`.

https://github.com/streamnative/kop/blob/cc4ba412c5e776242de3d7b4e0ff732e9bbf299a/kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/storage/PartitionLog.java#L177-L178

It will bring a problem that if `KafkaTopicManager#getTopic` took much time to complete, the `kop_server_PRODUCE_ENCODE` could be very large, while the actual encode time (`EntryFormatter#encode`) is not.

**Expected behavior**
We should split the current encode time interval into multiple parts to make the wait time for `getTopic` individual.

### PR DESCRIPTION
**Pull request by ZiyaoWei**
Fixes #1418

### Motivation

The metric PRODUCE_ENCODE isn't being calculated correctly. The starting timestamp used for the calculation is too early.

### Modifications

Add a new metric PENDING_TOPIC_LATENCY to record the latency before the pending topic future is called, and record the actual encoding time with PRODUCE_ENCODE.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### Documentation

Check the box below.

Need to update docs? 
  
- [x] `no-need-doc` 
  
  No functionality change

### CODE REVIEW

**Comment by ZiyaoWei** [comment]
Not sure how to best test this, also not sure if we should add documentation for it>

**Comment by Demogorgon314** [comment]
You can see the `MetricsProviderTest`. We can check the `PENDING_TOPIC_LATENCY ` is contained in response stats.

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KopServerStats.java [hunk 1]
```diff
@@ -51,6 +51,7 @@ public interface KopServerStats {
     /**
      * PRODUCE STATS.
      */
+    String PENDING_TOPIC_LATENCY = "PENDING_TOPIC_LATENCY";
```
Comment (Demogorgon314) [inline kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KopServerStats.java:54]: We also need to add the new metrics to `docs/reference-metrics.md`.

Comment (ZiyaoWei) [reply kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KopServerStats.java:54]: Thanks! Somehow missed it in my search.

**Review by Demogorgon314 — APPROVED** [review]

**Review by BewareMyPower — APPROVED** [review]

**Pull request merged**